pkg/dbmate/driver.go

```diff
@@ -17,6 +17,8 @@ type Driver interface {
 	SelectMigrations(*sql.DB, int) (map[string]bool, error)
 	InsertMigration(Transaction, string) error
 	DeleteMigration(Transaction, string) error
+	AcquireChangeLock(*sql.DB) (bool, error)
```

Comment:
This is a nit but I would probably call these methods `AcquireMigrationLock` and `ReleaseMigrationLock`, or maybe just `AcquireLock`? ChangeLock sounds a little weird to me.